js/hub.md

```diff
@@ -48,43 +36,208 @@ class MyClass {
 }
 ```
 
-In order to capture event updates, you need to implement `onHubCapsule` handler function in your listener class.
+**A note about onHubCapsule** In previous versions of Amplify capturing updates required you to implement an `onHubCapsule` handler function in your class and pass in `this` to the listen method. While still possible, this is no longer considered best practice and we have begun deprecating the method. Please define an explicit callback and pass it into the listen function (e.g. `Hub.listen('auth', this.myCallback)`) or use an anonymous function such as in the above example.
 {: .callout .callout--info}
 
-### Listening Authentication Events
+### Sending messages
+
+Sending events to different channels is done with the `dispatch` function:
+```javascript
+Hub.dispatch(
+    'DogsChannel', 
+    { 
+        event: 'buttonClick', 
+        data: {color:'blue'}, 
+        message:'' 
+});
+
+setTimeout(() => {
+  Hub.dispatch(
+      'CatsChannel', 
+      { 
+            event: 'drinkMilk', 
+            data: { 
+                breed: 'Persian',
+                age: 5
+            },
+            message: `The cat ${cat.name} has finished her milk`
+        });
+}, 5000)
+```
+`Hub.dispatch(channel: string, payload: HubPayload)` can be used to dispatch a `HubPayload` to a `channel`. The `channel` is a logical grouping for your organization while the `HubPayload` is a type defined as:
+```javascript
+export type HubPayload = {
+    event: string,
+    data?: any,
+    message?: string
+};
+```
 
-AWS Amplify Authentication module publishes in `auth` channel when 'signIn', 'signUp', and 'signOut' events happen. You can create your listener to listen and act upon those event notifications.
+The `event` field is recommended to be a small string without spaces such as `signIn` or `hang_up` as it's useful for checking payload groupings. The `data` field is a freeform structure which many times is used for larger JSON objects or custom data types. Finally while `message` is optional, we encourage you to use it as it is required when using a `RegExp` filtering with `Hub.listen()`.
+
+### Stop Listening
+
+Hub provides a way to stop listening for messages with `Hub.remove(channel: string | RegExp, listener: callback)`. This may be useful if you no longer need to receive messages in your application flow, as well as to avoid any memory leaks on low powered devices when you are sending large amounts of data through Hub on multiple channels.
+
+
+### Channels
+A channel is a logical group name that you use to organize messages and listen on. These are strings and completely up to you as the developer to define for dispatching or listening. However, while you can dispatch to any channel, ***Amplify protects certain channels*** and will flag a warning as sending unexpected payloads could have undesirable side effects (such as impacting authentication flows). The protected channels are currently:
+
+- core
```

Comment:
I think we should implement a namespace for our channels to make it even harder for our customers to use one of our channels by mistake.

For example, the channel names could be like this: `@aws-amplify/core`, `@aws-amplify/auth`, etc.

Reply:
then we could simply document that any channel that starts with `@aws-amplify/` is protected

Reply:
We talked through this during design as well and it seemed confusing especially for auth flows that are already using the current names. Doing the above would break existing apps too so it wouldn't be backwards compatible. We could look at it again as part of a major release though. Customers can still create independent Hub instances themselves or read/write to channels so that is essentially a namespace.